discovery-infra/tests/test_validations.py

```diff
@@ -0,0 +1,76 @@
+    @pytest.mark.regression
+    def test_basic_cluster_validations(self, api_client, cluster):
+        cluster_id = cluster().id
+
+        # Check initial validations
+        self.wait_until_cluster_is_pending_for_input(cluster_id, api_client)
+        orig_cluster = api_client.cluster_get(cluster_id)
+        self.assert_cluster_validation(orig_cluster, "configuration", "pull-secret-set", "success")
+        self.assert_cluster_validation(orig_cluster, "hosts-data", "all-hosts-are-ready-to-install", "success")
+        self.assert_cluster_validation(orig_cluster, "hosts-data", "sufficient-masters-count", "failure")
+        self.assert_cluster_validation(orig_cluster, "network", "machine-cidr-defined", "failure")
+        self.assert_cluster_validation(orig_cluster, "network", "machine-cidr-equals-to-calculated-cidr", "pending")
+        self.assert_cluster_validation(orig_cluster, "network", "api-vip-defined", "pending")
+        self.assert_cluster_validation(orig_cluster, "network", "api-vip-valid", "pending")
+        self.assert_cluster_validation(orig_cluster, "network", "ingress-vip-defined", "pending")
+        self.assert_cluster_validation(orig_cluster, "network", "ingress-vip-valid", "pending")
+        self.assert_cluster_validation(orig_cluster, "network", "dns-domain-defined", "success")
+        self.assert_cluster_validation(orig_cluster, "network", "cluster-cidr-defined", "success")
+        self.assert_cluster_validation(orig_cluster, "network", "service-cidr-defined", "success")
+        self.assert_cluster_validation(orig_cluster, "network", "no-cidrs-overlapping", "pending")
+        self.assert_cluster_validation(orig_cluster, "network", "network-prefix-valid", "success")
+        self.assert_cluster_validation(orig_cluster, "network", "ntp-server-configured", "success")
+
+        # Check base DNS domain
+        api_client.update_cluster(cluster_id, {"base_dns_domain": ""})
+        self.wait_for_cluster_validation(api_client, cluster_id, "network", "dns-domain-defined", ["failure"])
+        api_client.update_cluster(cluster_id, {"base_dns_domain": orig_cluster.base_dns_domain})
+        self.wait_for_cluster_validation(api_client, cluster_id, "network", "dns-domain-defined", ["success"])
+
+        # Check pull secret
+        api_client.update_cluster(cluster_id, {"pull_secret": ""})
+        self.wait_for_cluster_validation(api_client, cluster_id, "configuration", "pull-secret-set", ["failure"])
+        api_client.update_cluster(cluster_id, {"pull_secret": env_variables['pull_secret']})
+        self.wait_for_cluster_validation(api_client, cluster_id, "configuration", "pull-secret-set", ["success"])
+
+    def test_host_insufficient(self, api_client, cluster, nodes):
+        cluster_id = cluster().id
+        self.generate_and_download_image(cluster_id=cluster_id, api_client=api_client)
+
+        # Modify vCPU count on the first node to be insufficient
+        vcpu_node = nodes.nodes[0]
```

Comment:
use nodes.get_random_node(), then node.name

Reply:
How do I know that I won't get the same random node below, for ram_node?

Reply:
ho, i thought that node.nodes is also kind of a random